src/DataStructures/Tensor/Expressions/Normalize.hpp

```diff
@@ -0,0 +1,58 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions normalize_with_euclidean_metric and normalize
+
+#pragma once
+
+class DataVector;
+
+#include "DataStructures/Tensor/Expressions/Magnitude.hpp"
+#include "DataStructures/Tensor/Expressions/ScaleBy.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+
+/*!
+ * \ingroup Functions
+ * \brief normalize a rank-1 tensor assuming a Euclidean metric
+ *
+ * \returns the normalized rank-1 tensor
+ *
+ * \remark If you need to store the magnitude of the rank-1 tensor and the
+ * normalized tensor,  it is better to call euclidean_magnitude and scale_by
+ * separately, and not use this function.
+ *
+ * \details
+ * Scales the input tensor by its magnitude assuming a Euclidean metric.
+ *
+ */
+template <typename Index>
+Tensor<DataVector, Symmetry<1>, typelist<Index>>
+normalize_with_euclidean_metric(
+    const Tensor<DataVector, Symmetry<1>, typelist<Index>>& tensor) {
+  return scale_by(tensor, euclidean_magnitude(tensor));
+}
+
+/*!
+ * \ingroup Functions
+ * \brief normalize a rank-1 tensor with the given metric
+ *
+ * \returns the normalized rank-1 tensor
+ *
+ * \requires the metric to have the opposite valence of the input tensor
+ *
+ * \remark If you need to store the magnitude of the rank-1 tensor and the
+ * normalized tensor,  it is better to call euclidean_magnitude and scale_by
+ * separately, and not use this function.
+ *
+ * \details
+ * Scales the input tensor by its magnitude.
```

Comment:
This is ambiguous: the verb "to scale" applies to "scaling up" as well as "scaling down". (Oxford: "reduce or increase in size according to a common scale").

Could remove ambiguity by writing "Scales the input tensor by dividing by its magnitude.".
But this is just defining normalization, so becomes redundant with the function's name, so perhaps can be done without.